API/AddUser.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+	$inData = getRequestInfo();
+
+	$servername = "localhost";
+	$username = "root";
+	$password = "";
+	$dbname = "myDB";
+
+	// Create connection
+	$conn = new mysqli($servername, $username, $password, $dbname); 	
+	if( $conn->connect_error )
+	{
+		returnWithError( $conn->connect_error );
+	}
+	else
+	{
+		if userExists($conn, $inData['userName'])
+		{
+			returnWithError("Username Already Taken")
+		}
+		else
+		{
+			$hashed_password = password_hash($inData["password"], PASSWORD_DEFAULT);
+			
+			$stmt = $conn->prepare("INSERT INTO Users (ID, firstName, lastName, userName, password, dateCreated) VALUES (NULL, ?, ?, ?, ?, NULL)");
```

Comment:
Should use DEFAULT for ID and DateCreated as those columns are auto-generated.
```suggestion
			$stmt = $conn->prepare("INSERT INTO Users (ID, firstName, lastName, userName, password, dateCreated) VALUES (DEFAULT, ?, ?, ?, ?, DEFAULT)");
```